### PR TITLE
Species Damage Modifiers Affect Targeted External Organ Damage

### LINKED ISSUE #9533: Golems Take Burn Damage From Shocked Grilles
This is more of a self-reminder than anything, I intend to fix it.
Apparently it's because electric shocks manually alter the mob's burn damage rather than using `adjustBurnLoss()`.

### PR DESCRIPTION
**Pull request by Shadowmech88**
`/datum/organ/external/proc/take_damage()` now takes the burn_mod and brute_mod of the organ's owner's species into account when applying damage.
The most obvious effect of this is that now diona will take increased burn damage from electrical shocks according to their burn_mod, and golems will take decreased burn damage from electrical shocks according to their burn_mod.

Fixes #9533.

### CODE REVIEW

**Comment by 9600bauds** [comment]
:+1:

File: code/game/objects/structures/electricchair.dm [hunk 1]
```diff
@@ -73,6 +73,12 @@
 		M.Stun(60)
 		M.Jitter(60)
 		visible_message("<span class='danger'>The electric chair went off!</span>", "<span class='danger'>You hear a deep sharp shock!</span>")
+		var/mob/living/carbon/human/H = null
+		if(ishuman(M))
+			H = M
+			if(isgolem(H)) //golems can't be burned
```
Comment (9600bauds) [inline code/game/objects/structures/electricchair.dm]: Wouldn't it be better and less snowflakey to just handle golems in proc/burn_skin()?

File: html/changelogs/Shadowmech88.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+author: Shadowmech88
+delete-after: True
+changes: 
+- tweak: Species damage modifiers are now applied when individual external organs take damage.
```
Comment (9600bauds) [inline html/changelogs/Shadowmech88.yml]: Probably use layman's terms here like in the PR

File: code/game/objects/structures/electricchair.dm (hunk 1 above)
Comment (Shadowmech88) [inline code/game/objects/structures/electricchair.dm]: I actually just had a look and `burn_skin()` calls `/organ/external/proc/take_damage()` anyway, so I'm getting two birds with one stone.

**Comment by ihadtoregisterforthis** [comment]
muh gollum powercreep

**Comment by kol1th** [comment]
:+1:

**Comment by gbasood** [comment]
@clusterfack @PJB3005 pls review

**Pull request merged**